### ISSUE TITLE
Can't select NanoFlow as rule

### ISSUE THREAD

**Issue by 0daven** [tieniber/CellStyler#5]
Adding this widget to a Mendix 7.13.1 does not give the option to select a NanoFlow, although it works in the test package provided.

![image](https://user-images.githubusercontent.com/26849638/42283221-bedb39c2-7fa0-11e8-8bb7-baec571aefc3.png)

Am I missing something?

**Comment by tieniber** [comment]
Did you download the version from the GitHub repository or the App Store?
The App Store version is older and does not have nanoflow support yet.

**Comment by 0daven** [comment]
I downloaded from App store.

If I try to use the one from Github it crashes the Modeler after first saving options.

**Comment by tieniber** [comment]
Can you share the modeler error (if any?)

I’ll also try to reproduce this myself in a new project. If that doesn’t
work, perhaps you can share you project with me.

**Comment by 0daven** [comment]
This is the modeler output (happens as soon as I try to save any rules with a NanoFlow):

System.InvalidOperationException: Unknown property type 'Nanoflow'.
   at Mendix.Modeler.WebUI.Forms.Widgets.CustomWidgets.WidgetValue.ToString() in C:\jenkins\workspace\AppStudio3.0-Build\modeler\Mendix.Modeler.WebUI\Forms\Widgets\CustomWidgets\WidgetValue.cs:line 489
   at Mendix.Modeler.Utility.Grids.GridColumn`2.Mendix.Modeler.Utility.Grids.IGridColumn.GetValue(Object obj) in C:\jenkins\workspace\AppStudio3.0-Build\modeler\Mendix.Modeler.Utility.Gui\Grids\GridColumn.cs:line 105
   at Mendix.Modeler.Utility.Grids.Grid.RefreshContents() in C:\jenkins\workspace\AppStudio3.0-Build\modeler\Mendix.Modeler.Utility.Gui\Grids\Grid.cs:line 398
   at Mendix.Modeler.Utility.Grids.Grid.DoRebuild() in C:\jenkins\workspace\AppStudio3.0-Build\modeler\Mendix.Modeler.Utility.Gui\Grids\Grid.cs:line 128
   at Mendix.Modeler.Utility.Grids.Grid.WithoutRaisingEventsDo(Action code) in C:\jenkins\workspace\AppStudio3.0-Build\modeler\Mendix.Modeler.Utility.Gui\Grids\Grid.cs:line 584
   at Mendix.Modeler.Utility.Grids.Grid.Rebuild() in C:\jenkins\workspace\AppStudio3.0-Build\modeler\Mendix.Modeler.Utility.Gui\Grids\Grid.cs:line 93
   at Mendix.Modeler.Utility.Grids.Grid.RebuildWithEvents() in C:\jenkins\workspace\AppStudio3.0-Build\modeler\Mendix.Modeler.Utility.Gui\Grids\Grid.cs:line 231
   at Mendix.Modeler.Utility.Grids.GridConfig`1.<>c__DisplayClass58_0.<AddToolNewButton>b__0(List`1 objs) in C:\jenkins\workspace\AppStudio3.0-Build\modeler\Mendix.Modeler.Utility.Gui\Grids\GridConfig.cs:line 271
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripButton.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at Mendix.Modeler.Utility.ToolStripEx.WndProc(Message& m) in C:\jenkins\workspace\AppStudio3.0-Build\modeler\Mendix.Modeler.Utility.Gui\Toolstrip\ToolStripEx.cs:line 142
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by tieniber** [comment]
Ok, I've confirmed the issue. It seems that the modeler does not like when a nanoflow can be selected as one of the first 3 properties in a list. I've submitted a ticket to Mendix R&D to fix this. In the mean time, the workaround will be to make the nanoflow the 4th option instead of the third.

For anyone who hasn't imported this widget into their project yet, the latest version committed in this repository should work fine. Unfortunately, the modeler has a kind of stickiness to the order of properties in custom widgets, and so for you project you'll need to download this file, unzip and replace your project's current CellStyler.mpk with this one.

[CellStyler.mpk.zip](https://github.com/tieniber/CellStyler/files/2167314/CellStyler.mpk.zip)

Once you do that, press F4 in the modeler and then right-click on your Cell Styler widget and choose "update widget". This copy of the widget does not have the Nanoflow or JS properties in it. Open the widget properties and ensure this is the case.

Then, download the latest widget from this repository and repeat the steps above. That should get it working for you.
